File: Syncfusion_Sample/Windows/Chart.Windows/Samples/2.0/User Interaction/Chart Interactive Cursor/CS/FormMain.cs
#region Copyright Syncfusion Inc. 2001 - 2006
//
//  Copyright Syncfusion Inc. 2001 - 2006. All rights reserved.
//
//  Use of this code is subject to the terms of our license.
//  A copy of the current license can be obtained at any time by e-mailing
//  licensing@syncfusion.com. Any infringement will be prosecuted under
//  applicable laws. 
//
#endregion

using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using Syncfusion.Windows.Forms.Chart;
using Syncfusion.Drawing;

namespace Syncfusion.Windows.Forms.Chart.Samples
{
    public class FormMain : Syncfusion.Windows.Forms.Office2007Form
    {
        private IContainer components = null;
        #region Private Members

        private System.Windows.Forms.Panel panel1;
        private ChartInteractiveCursor cursor1;
        private Syncfusion.Windows.Forms.ColorPickerButton colorPickerButton1;
        private Label label2;
        private Label label3;
        private ComboBox comboBox2;
        private Label label4;
        private ComboBox comboBox1;
        private Label label1;
        private CheckBox checkBox1;
        private ChartControl chartControl1;
        #endregion

        #region Constructor, Main and Dispose
        public FormMain()
        {
            InitializeComponent();

        }
        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main()
        {
            Application.EnableVisualStyles();
            Application.Run(new FormMain());
        }
        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(disposing);
        }
        #endregion

        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            System.Drawing.StringFormat stringFormat1 = new System.Drawing.StringFormat();
            System.Drawing.StringFormat stringFormat2 = new System.Drawing.StringFormat();
            System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(FormMain));
            this.chartControl1 = new Syncfusion.Windows.Forms.Chart.ChartControl();
            this.panel1 = new System.Windows.Forms.Panel();
            this.comboBox2 = new System.Windows.Forms.ComboBox();
            this.label4 = new System.Windows.Forms.Label();
            this.comboBox1 = new System.Windows.Forms.ComboBox();
            this.label1 = new System.Windows.Forms.Label();
            this.checkBox1 = new System.Windows.Forms.CheckBox();
            this.label2 = new System.Windows.Forms.Label();
            this.colorPickerButton1 = new Syncfusion.Windows.Forms.ColorPickerButton();
            this.label3 = new System.Windows.Forms.Label();
            this.panel1.SuspendLayout();
            this.SuspendLayout();
            // 
            // chartControl1
            // 
            this.chartControl1.ChartArea.BackInterior = new Syncfusion.Drawing.BrushInfo(Syncfusion.Drawing.GradientStyle.Vertical, System.Drawing.Color.Transparent, System.Drawing.Color.Transparent);
            this.chartControl1.ChartArea.CursorLocation = new System.Drawing.Point(0, 0);
            this.chartControl1.ChartArea.CursorReDraw = false;
            this.chartControl1.IsWindowLess = false;
            // 
            // 
            // 
            this.chartControl1.Legend.Location = new System.Drawing.Point(382, 75);
            this.chartControl1.Localize = null;
            this.chartControl1.Location = new System.Drawing.Point(8, 12);
            this.chartControl1.Name = "chartControl1";
            this.chartControl1.PrimaryXAxis.Crossing = double.NaN;
            stringFormat1.Alignment = System.Drawing.StringAlignment.Near;
            stringFormat1.HotkeyPrefix = System.Drawing.Text.HotkeyPrefix.None;
            stringFormat1.LineAlignment = System.Drawing.StringAlignment.Near;
            stringFormat1.Trimming = System.Drawing.StringTrimming.Character;
            this.chartControl1.PrimaryXAxis.LabelStringFormat = stringFormat1;
            this.chartControl1.PrimaryXAxis.Margin = true;
            this.chartControl1.PrimaryYAxis.Crossing = double.NaN;
            stringFormat2.Alignment = System.Drawing.StringAlignment.Far;
            stringFormat2.HotkeyPrefix = System.Drawing.Text.HotkeyPrefix.None;
            stringFormat2.LineAlignment = System.Drawing.StringAlignment.Near;
            stringFormat2.Trimming = System.Drawing.StringTrimming.Character;
            this.chartControl1.PrimaryYAxis.LabelStringFormat = stringFormat2;
            this.chartControl1.PrimaryYAxis.Margin = true;
            this.chartControl1.Size = new System.Drawing.Size(491, 345);
            this.chartControl1.TabIndex = 1;
            this.chartControl1.Text = "chartControl1";
            // 
            // 
            // 
            this.chartControl1.Title.Name = "Def_title";
            this.chartControl1.Title.Text = "chartControl1";
            this.chartControl1.Titles.Add(this.chartControl1.Title);
            this.chartControl1.ZoomOutIncrement = 1.5D;
            // 
            // panel1
            // 
            this.panel1.BackColor = System.Drawing.Color.White;
            this.panel1.Controls.Add(this.comboBox2);
            this.panel1.Controls.Add(this.label4);
            this.panel1.Controls.Add(this.comboBox1);
            this.panel1.Controls.Add(this.label1);
            this.panel1.Controls.Add(this.checkBox1);
            this.panel1.Controls.Add(this.label2);
            this.panel1.Controls.Add(this.colorPickerButton1);
            this.panel1.Dock = System.Windows.Forms.DockStyle.Bottom;
            this.panel1.Location = new System.Drawing.Point(0, 381);
            this.panel1.Name = "panel1";
            this.panel1.Size = new System.Drawing.Size(614, 77);
            this.panel1.TabIndex = 2;
            // 
            // comboBox2
            // 
            this.comboBox2.FormattingEnabled = true;
            this.comboBox2.Items.AddRange(new object[] {
            "100",
            "150",
            "200"});
            this.comboBox2.Location = new System.Drawing.Point(250, 41);
            this.comboBox2.Name = "comboBox2";
            this.comboBox2.Size = new System.Drawing.Size(121, 21);
            this.comboBox2.TabIndex = 15;
            this.comboBox2.SelectedIndexChanged += new System.EventHandler(this.comboBox2_SelectedIndexChanged);
            // 
            // label4
            // 
            this.label4.AutoSize = true;
            this.label4.Location = new System.Drawing.Point(183, 41);
            this.label4.Name = "label4";
            this.label4.Size = new System.Drawing.Size(49, 13);
            this.label4.TabIndex = 14;
            this.label4.Text = "YInterval";
            // 
            // comboBox1
            // 
            this.comboBox1.FormattingEnabled = true;
            this.comboBox1.Items.AddRange(new object[] {
            "2",
            "4",
            "6"});
            this.comboBox1.Location = new System.Drawing.Point(250, 4);
            this.comboBox1.Name = "comboBox1";
            this.comboBox1.Size = new System.Drawing.Size(121, 21);
            this.comboBox1.TabIndex = 13;
            this.comboBox1.SelectedIndexChanged += new System.EventHandler(this.comboBox1_SelectedIndexChanged);
            // 
            // label1
            // 
            this.label1.AutoSize = true;
            this.label1.Location = new System.Drawing.Point(183, 8);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(49, 13);
            this.label1.TabIndex = 12;
            this.label1.Text = "XInterval";
            // 
            // checkBox1
            // 
            this.checkBox1.AutoSize = true;
            this.checkBox1.Checked = true;
            this.checkBox1.CheckState = System.Windows.Forms.CheckState.Checked;
            this.checkBox1.Location = new System.Drawing.Point(34, 8);
            this.checkBox1.Name = "checkBox1";
            this.checkBox1.Size = new System.Drawing.Size(116, 17);
            this.checkBox1.TabIndex = 11;
            this.checkBox1.Text = "MoveTo ChartArea";
            this.checkBox1.UseVisualStyleBackColor = true;
            this.checkBox1.CheckedChanged += new System.EventHandler(this.checkBox1_CheckedChanged);
            // 
            // label2
            // 
            this.label2.Location = new System.Drawing.Point(478, 5);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(71, 16);
            this.label2.TabIndex = 10;
            this.label2.Text = "Cursor Color";
            // 
            // colorPickerButton1
            // 
            this.colorPickerButton1.BackColor = System.Drawing.Color.Black;
            this.colorPickerButton1.ColorUISize = new System.Drawing.Size(208, 230);
            this.colorPickerButton1.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            this.colorPickerButton1.Location = new System.Drawing.Point(566, 0);
            this.colorPickerButton1.Name = "colorPickerButton1";
            this.colorPickerButton1.SelectedAsBackcolor = true;
            this.colorPickerButton1.SelectedColor = System.Drawing.Color.Black;
            this.colorPickerButton1.Size = new System.Drawing.Size(24, 24);
            this.colorPickerButton1.TabIndex = 2;
            this.colorPickerButton1.UseVisualStyleBackColor = false;
            this.colorPickerButton1.ColorSelected += new System.EventHandler(this.Color_selected);
            // 
            // label3
            // 
            this.label3.BackColor = System.Drawing.Color.Transparent;
            this.label3.Font = new System.Drawing.Font("Verdana", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(161)));
            this.label3.Location = new System.Drawing.Point(54, 365);
            this.label3.Name = "label3";
            this.label3.Size = new System.Drawing.Size(457, 13);
            this.label3.TabIndex = 2;
            this.label3.Text = "Click on the interactive cursor line and drag the chart area or series points";
            this.label3.TextAlign = System.Drawing.ContentAlignment.MiddleCenter;
            // 
            // FormMain
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.BackColor = System.Drawing.Color.White;
            this.ClientSize = new System.Drawing.Size(614, 458);
            this.Controls.Add(this.label3);
            this.Controls.Add(this.chartControl1);
            this.Controls.Add(this.panel1);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
            this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
            this.MaximizeBox = false;
            this.Name = "FormMain";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen;
            this.Text = "Chart Interactive Cursor";
            this.Load += new System.EventHandler(this.FormMain_Load);
            this.panel1.ResumeLayout(false);
            this.panel1.PerformLayout();
            this.ResumeLayout(false);

        }
        #endregion

        #region Form Load
        private void FormMain_Load(object sender, EventArgs e)
        {
            InitializeChartData();
            this.comboBox1.SelectedIndex = 0;
            this.comboBox2.SelectedIndex = 0;
            
            this.label1.Enabled = true;
            this.label4.Enabled = true;
            this.comboBox1.Enabled = true;
            this.comboBox2.Enabled = true;
            ChartAppearance.ApplyChartStyles(this.chartControl1);
            //InitializeControlSettings();
        }
        #endregion

        #region Helper Methods

        #region InitializeChartData
        /// <summary>
        /// Initializes the ChartControl's data and sets the Chart type
        /// </summary>
        protected void InitializeChartData()
        {
            ChartSeries series1 = new ChartSeries();
            series1.Name = "Server 1";
            series1.Type = ChartSeriesType.Line;
            series1.Text = series1.Name;
            int[] r1 = new int[] { 256, 491, 382, 437, 321, 234, 425, 257, 382, 296, 437, 451, 490, 366, 422, 361, 185, 384, 461, 258 };
            for (int i = 2; i < 20; i = i + 2)
                series1.Points.Add(i, r1[i]);

            series1.PrepareStyle += new ChartPrepareStyleInfoHandler(ChartControlSeries_PrepareStyle);

            this.chartControl1.Series.Add(series1);
            this.chartControl1.Text = "Chart Interactive cursor";
            cursor1 = new ChartInteractiveCursor(this.chartControl1.Series[0]);
            this.chartControl1.ChartArea.InteractiveCursors.Add(cursor1);
            cursor1.Color = Color.Black;
            this.chartControl1.ChartArea.InteractiveCursors[0].XPosition = 2;
            this.chartControl1.ChartArea.InteractiveCursors[0].YPosition = 2;
            this.chartControl1.ChartArea.InteractiveCursors[0].MoveToChartArea = this.checkBox1.Checked;
            if (this.chartControl1.ChartArea.InteractiveCursors[0].MoveToChartArea)
            {
                this.chartControl1.ChartArea.InteractiveCursors[0].XInterval = 2;
                this.chartControl1.ChartArea.InteractiveCursors[0].YInterval = 50;
            }
        }
        #endregion
        #endregion

        #region Events
        #region PrepareStyle
        /// <summary>
        /// Handler for the PrepareStyle event of ChartSeries. 
        /// </summary>
        /// <param name="sender">The ChartSeries object that raises this event.</param>
        /// <param name="args">The event data. Can be modified to change series appearance.</param>
        protected void ChartControlSeries_PrepareStyle(object sender, ChartPrepareStyleInfoEventArgs args)
        {
            // Style formatting using a callback. You can apply the same settings directly on the series style on the
            // point styles.
            ChartSeries series = sender as ChartSeries;
            if (series != null)
            {
                args.Style.DisplayText = true;
                args.Style.TextFormat = "{0}";
                args.Style.TextOrientation = ChartTextOrientation.Up;
                args.Style.Symbol.Shape = ChartSymbolShape.Circle;
                args.Style.Symbol.Size = new Size(10, 10);
                args.Style.Symbol.Color = Color.White;
                args.Style.Symbol.Border.Color = Color.FromArgb(129, 115, 27);
                args.Style.Symbol.Border.Width = 2;
            }
        }
        #endregion


        private void Color_selected(object sender, System.EventArgs e)
        {
            cursor1.Color = this.colorPickerButton1.SelectedColor;

            this.chartControl1.Redraw(true);
        }

       
        #endregion

        private void checkBox1_CheckedChanged(object sender, EventArgs e)
        {
            if (this.checkBox1.Checked && this.chartControl1.ChartArea.InteractiveCursors.Count >0)
            {
                this.chartControl1.ChartArea.InteractiveCursors[0].MoveToChartArea = this.checkBox1.Checked;
                this.label1.Enabled = true;
                this.label4.Enabled = true;
                this.comboBox1.Enabled = true;
                this.comboBox2.Enabled = true;
            }
            else
            {
                if (this.chartControl1.ChartArea.InteractiveCursors.Count > 0)
                {
                    this.chartControl1.ChartArea.InteractiveCursors[0].MoveToChartArea = this.checkBox1.Checked;
                }
                this.label1.Enabled = false;
                this.label4.Enabled = false;
                this.comboBox1.Enabled = false;
                this.comboBox2.Enabled = false;
            }
            this.chartControl1.Redraw(true);
        }

        private void comboBox1_SelectedIndexChanged(object sender, EventArgs e)
        {
            this.chartControl1.ChartArea.InteractiveCursors[0].XInterval = Convert.ToDouble(this.comboBox1.SelectedItem);
        }

        private void comboBox2_SelectedIndexChanged(object sender, EventArgs e)
        {

            this.chartControl1.ChartArea.InteractiveCursors[0].YInterval = Convert.ToDouble(this.comboBox2.SelectedItem);
        }
    }
}

File: Syncfusion_Sample/Windows/Chart.Windows/Samples/2.0/User Interaction/Chart Interactive Cursor/VB/FormMain.vb
#Region "Copyright Syncfusion Inc. 2001 - 2006"
'
'  Copyright Syncfusion Inc. 2001 - 2006. All rights reserved.
'
'  Use of this code is subject to the terms of our license.
'  A copy of the current license can be obtained at any time by e-mailing
'  licensing@syncfusion.com. Any infringement will be prosecuted under
'  applicable laws. 
'
#End Region


Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports Syncfusion.Windows.Forms.Chart
Imports Syncfusion.Drawing

Namespace Syncfusion.Windows.Forms.Chart.Samples
	Public Class FormMain
		Inherits Syncfusion.Windows.Forms.Office2007Form
		Private components As IContainer
		#Region "Private Members"

		Private panel1 As System.Windows.Forms.Panel
		Private cursor1 As ChartInteractiveCursor
		Private WithEvents colorPickerButton1 As Syncfusion.Windows.Forms.ColorPickerButton
        Private label2 As Label
        Private label3 As Label
        Private WithEvents comboBox2 As System.Windows.Forms.ComboBox
        Private WithEvents label4 As System.Windows.Forms.Label
        Private WithEvents comboBox1 As System.Windows.Forms.ComboBox
        Private WithEvents label1 As System.Windows.Forms.Label
        Private WithEvents checkBox1 As System.Windows.Forms.CheckBox
        Private chartControl1 As ChartControl
#End Region

#Region "Constructor, Main and Dispose"
        Public Sub New()
            InitializeComponent()

        End Sub
        ''' <summary>
        ''' The main entry point for the application.
        ''' </summary>
        <STAThread()> _
        Shared Sub Main()
            Application.EnableVisualStyles()
            Application.Run(New FormMain())
        End Sub
        ''' <summary>
        ''' Clean up any resources being used.
        ''' </summary>
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not components Is Nothing Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub
#End Region

#Region "Windows Form Designer generated code"
        ''' <summary>
        ''' Required method for Designer support - do not modify
        ''' the contents of this method with the code editor.
        ''' </summary>
        Private Sub InitializeComponent()
            Dim StringFormat1 As System.Drawing.StringFormat = New System.Drawing.StringFormat()
            Dim StringFormat2 As System.Drawing.StringFormat = New System.Drawing.StringFormat()
            Dim resources As System.ComponentModel.ComponentResourceManager = New System.ComponentModel.ComponentResourceManager(GetType(FormMain))
            Me.chartControl1 = New Syncfusion.Windows.Forms.Chart.ChartControl()
            Me.panel1 = New System.Windows.Forms.Panel()
            Me.comboBox2 = New System.Windows.Forms.ComboBox()
            Me.label4 = New System.Windows.Forms.Label()
            Me.comboBox1 = New System.Windows.Forms.ComboBox()
            Me.label1 = New System.Windows.Forms.Label()
            Me.checkBox1 = New System.Windows.Forms.CheckBox()
            Me.label2 = New System.Windows.Forms.Label()
            Me.colorPickerButton1 = New Syncfusion.Windows.Forms.ColorPickerButton()
            Me.label3 = New System.Windows.Forms.Label()
            Me.panel1.SuspendLayout()
            Me.SuspendLayout()
            '
            'chartControl1
            '
            Me.chartControl1.ChartArea.BackInterior = New Syncfusion.Drawing.BrushInfo(Syncfusion.Drawing.GradientStyle.Vertical, System.Drawing.Color.Transparent, System.Drawing.Color.Transparent)
            Me.chartControl1.ChartArea.CursorLocation = New System.Drawing.Point(0, 0)
            Me.chartControl1.ChartArea.CursorReDraw = False
            Me.chartControl1.IsWindowLess = False
            '
            '
            '
            Me.chartControl1.Legend.Location = New System.Drawing.Point(389, 75)
            Me.chartControl1.Localize = Nothing
            Me.chartControl1.Location = New System.Drawing.Point(9, 12)
            Me.chartControl1.Name = "chartControl1"
            Me.chartControl1.PrimaryXAxis.Crossing = Double.NaN
            StringFormat1.Alignment = System.Drawing.StringAlignment.Near
            StringFormat1.HotkeyPrefix = System.Drawing.Text.HotkeyPrefix.None
            StringFormat1.LineAlignment = System.Drawing.StringAlignment.Near
            StringFormat1.Trimming = System.Drawing.StringTrimming.Character
            Me.chartControl1.PrimaryXAxis.LabelStringFormat = StringFormat1
            Me.chartControl1.PrimaryXAxis.Margin = True
            Me.chartControl1.PrimaryYAxis.Crossing = Double.NaN
            StringFormat2.Alignment = System.Drawing.StringAlignment.Far
            StringFormat2.HotkeyPrefix = System.Drawing.Text.HotkeyPrefix.None
            StringFormat2.LineAlignment = System.Drawing.StringAlignment.Near
            StringFormat2.Trimming = System.Drawing.StringTrimming.Character
            Me.chartControl1.PrimaryYAxis.LabelStringFormat = StringFormat2
            Me.chartControl1.PrimaryYAxis.Margin = True
            Me.chartControl1.Size = New System.Drawing.Size(498, 345)
            Me.chartControl1.TabIndex = 1
            Me.chartControl1.Text = "chartControl1"
            '
            '
            '
            Me.chartControl1.Title.Name = "Def_title"
            Me.chartControl1.Title.Text = "chartControl1"
            Me.chartControl1.Titles.Add(Me.chartControl1.Title)
            Me.chartControl1.ZoomOutIncrement = 1.5R
            '
            'panel1
            '
            Me.panel1.BackColor = System.Drawing.Color.White
            Me.panel1.Controls.Add(Me.comboBox2)
            Me.panel1.Controls.Add(Me.label4)
            Me.panel1.Controls.Add(Me.comboBox1)
            Me.panel1.Controls.Add(Me.label1)
            Me.panel1.Controls.Add(Me.checkBox1)
            Me.panel1.Controls.Add(Me.label2)
            Me.panel1.Controls.Add(Me.colorPickerButton1)
            Me.panel1.Dock = System.Windows.Forms.DockStyle.Bottom
            Me.panel1.Location = New System.Drawing.Point(0, 381)
            Me.panel1.Name = "panel1"
            Me.panel1.Size = New System.Drawing.Size(604, 69)
            Me.panel1.TabIndex = 2
            '
            'comboBox2
            '
            Me.comboBox2.FormattingEnabled = True
            Me.comboBox2.Items.AddRange(New Object() {"100", "150", "200"})
            Me.comboBox2.Location = New System.Drawing.Point(228, 41)
            Me.comboBox2.Name = "comboBox2"
            Me.comboBox2.Size = New System.Drawing.Size(121, 21)
            Me.comboBox2.TabIndex = 20
            '
            'label4
            '
            Me.label4.AutoSize = True
            Me.label4.Location = New System.Drawing.Point(161, 41)
            Me.label4.Name = "label4"
            Me.label4.Size = New System.Drawing.Size(49, 13)
            Me.label4.TabIndex = 19
            Me.label4.Text = "YInterval"
            '
            'comboBox1
            '
            Me.comboBox1.FormattingEnabled = True
            Me.comboBox1.Items.AddRange(New Object() {"2", "4", "6"})
            Me.comboBox1.Location = New System.Drawing.Point(228, 4)
            Me.comboBox1.Name = "comboBox1"
            Me.comboBox1.Size = New System.Drawing.Size(121, 21)
            Me.comboBox1.TabIndex = 18
            '
            'label1
            '
            Me.label1.AutoSize = True
            Me.label1.Location = New System.Drawing.Point(161, 8)
            Me.label1.Name = "label1"
            Me.label1.Size = New System.Drawing.Size(49, 13)
            Me.label1.TabIndex = 17
            Me.label1.Text = "XInterval"
            '
            'checkBox1
            '
            Me.checkBox1.AutoSize = True
            Me.checkBox1.Checked = True
            Me.checkBox1.CheckState = System.Windows.Forms.CheckState.Checked
            Me.checkBox1.Location = New System.Drawing.Point(12, 8)
            Me.checkBox1.Name = "checkBox1"
            Me.checkBox1.Size = New System.Drawing.Size(116, 17)
            Me.checkBox1.TabIndex = 16
            Me.checkBox1.Text = "MoveTo ChartArea"
            Me.checkBox1.UseVisualStyleBackColor = True
            '
            'label2
            '
            Me.label2.Location = New System.Drawing.Point(407, 7)
            Me.label2.Name = "label2"
            Me.label2.Size = New System.Drawing.Size(71, 16)
            Me.label2.TabIndex = 10
            Me.label2.Text = "Cursor Color"
            '
            'colorPickerButton1
            '
            Me.colorPickerButton1.BackColor = System.Drawing.Color.Black
            Me.colorPickerButton1.ColorUISize = New System.Drawing.Size(208, 230)
            Me.colorPickerButton1.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            Me.colorPickerButton1.Location = New System.Drawing.Point(484, 1)
            Me.colorPickerButton1.Name = "colorPickerButton1"
            Me.colorPickerButton1.SelectedAsBackcolor = True
            Me.colorPickerButton1.SelectedColor = System.Drawing.Color.Black
            Me.colorPickerButton1.Size = New System.Drawing.Size(24, 24)
            Me.colorPickerButton1.TabIndex = 2
            Me.colorPickerButton1.UseVisualStyleBackColor = False
            '
            'label3
            '
            Me.label3.BackColor = System.Drawing.Color.White
            Me.label3.Font = New System.Drawing.Font("Verdana", 8.25!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(161, Byte))
            Me.label3.Location = New System.Drawing.Point(57, 360)
            Me.label3.Name = "label3"
            Me.label3.Size = New System.Drawing.Size(430, 14)
            Me.label3.TabIndex = 2
            Me.label3.Text = "Click on the interactive cursor line and drag to chart area or series points"
            Me.label3.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
            '
            'FormMain
            '
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.BackColor = System.Drawing.Color.White
            Me.ClientSize = New System.Drawing.Size(604, 450)
            Me.Controls.Add(Me.label3)
            Me.Controls.Add(Me.chartControl1)
            Me.Controls.Add(Me.panel1)
            Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
            Me.Icon = CType(resources.GetObject("$this.Icon"), System.Drawing.Icon)
            Me.MaximizeBox = False
            Me.Name = "FormMain"
            Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
            Me.Text = "Chart Interactive Cursor"
            Me.panel1.ResumeLayout(False)
            Me.panel1.PerformLayout()
            Me.ResumeLayout(False)

        End Sub
#End Region

#Region "Form Load"
        Private Sub FormMain_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            InitializeChartData()
            Me.comboBox1.SelectedIndex = 0
            Me.comboBox2.SelectedIndex = 0

            Me.label1.Enabled = True
            Me.label4.Enabled = True
            Me.comboBox1.Enabled = True
            Me.comboBox2.Enabled = True
            ChartAppearance.ApplyChartStyles(Me.chartControl1)

        End Sub
#End Region

#Region "Helper Methods"

#Region "InitializeChartData"
        ''' <summary>
        ''' Initializes the ChartControl's data and sets the Chart type
        ''' </summary>
        Protected Sub InitializeChartData()
            Dim series1 As ChartSeries = New ChartSeries()
            series1.Name = "Server 1"
            series1.Type = ChartSeriesType.Line
            series1.Text = series1.Name
            Dim r1 As Integer() = New Integer() {256, 491, 382, 437, 321, 234, 425, 257, 382, 296, 437, 451, 490, 366, 422, 361, 185, 384, 461, 258}
            For i As Integer = 2 To 19 Step 2
                series1.Points.Add(i, r1(i))
            Next i

            AddHandler series1.PrepareStyle, AddressOf ChartControlSeries_PrepareStyle

            Me.chartControl1.Series.Add(series1)
            cursor1 = New ChartInteractiveCursor(Me.chartControl1.Series(0))
            Me.chartControl1.ChartArea.InteractiveCursors.Add(cursor1)
            cursor1.Color = Color.Black
            Me.chartControl1.ShowLegend = False
            Me.chartControl1.ChartArea.InteractiveCursors(0).XPosition = 2
            Me.chartControl1.ChartArea.InteractiveCursors(0).YPosition = 5
            Me.chartControl1.ChartArea.InteractiveCursors(0).MoveToChartArea = Me.checkBox1.Checked
            If Me.chartControl1.ChartArea.InteractiveCursors(0).MoveToChartArea Then
                Me.chartControl1.ChartArea.InteractiveCursors(0).XInterval = 2
                Me.chartControl1.ChartArea.InteractiveCursors(0).YInterval = 50
            End If
        End Sub
#End Region
#End Region

#Region "Events"
#Region "PrepareStyle"
        ''' <summary>
        ''' Handler for the PrepareStyle event of ChartSeries. 
        ''' </summary>
        ''' <param name="sender">The ChartSeries object that raises this event.</param>
        ''' <param name="args">The event data. Can be modified to change series appearance.</param>
        Protected Sub ChartControlSeries_PrepareStyle(ByVal sender As Object, ByVal args As ChartPrepareStyleInfoEventArgs)
            ' Style formatting using a callback. You can apply the same settings directly on the series style on the
            ' point styles.
            Dim series As ChartSeries = TryCast(sender, ChartSeries)
            If Not series Is Nothing Then
                args.Style.DisplayText = True
                args.Style.TextFormat = "{0}"
                args.Style.TextOrientation = ChartTextOrientation.Up
                args.Style.Symbol.Shape = ChartSymbolShape.Circle
                args.Style.Symbol.Size = New Size(10, 10)
                args.Style.Symbol.Color = Color.White
                args.Style.Symbol.Border.Color = Color.FromArgb(129, 115, 27)
                args.Style.Symbol.Border.Width = 2
            End If
        End Sub
#End Region


        Private Sub Color_selected(ByVal sender As Object, ByVal e As System.EventArgs) Handles colorPickerButton1.ColorSelected
            cursor1.Color = Me.colorPickerButton1.SelectedColor

            Me.chartControl1.Redraw(True)
        End Sub

       
#End Region

        Private Sub checkBox1_CheckedChanged(sender As System.Object, e As System.EventArgs) Handles checkBox1.CheckedChanged
            If Me.checkBox1.Checked AndAlso Me.chartControl1.ChartArea.InteractiveCursors.Count > 0 Then
                Me.chartControl1.ChartArea.InteractiveCursors(0).MoveToChartArea = Me.checkBox1.Checked
                Me.label1.Enabled = True
                Me.label4.Enabled = True
                Me.comboBox1.Enabled = True
                Me.comboBox2.Enabled = True
            Else
                If Me.chartControl1.ChartArea.InteractiveCursors.Count > 0 Then
                    Me.chartControl1.ChartArea.InteractiveCursors(0).MoveToChartArea = Me.checkBox1.Checked
                End If
                Me.label1.Enabled = False
                Me.label4.Enabled = False
                Me.comboBox1.Enabled = False
                Me.comboBox2.Enabled = False
            End If
            Me.chartControl1.Redraw(True)
        End Sub

        Private Sub comboBox1_SelectedIndexChanged(sender As System.Object, e As System.EventArgs) Handles comboBox1.SelectedIndexChanged
            Me.chartControl1.ChartArea.InteractiveCursors(0).XInterval = Convert.ToDouble(Me.comboBox1.SelectedItem)
        End Sub

        Private Sub comboBox2_SelectedIndexChanged(sender As System.Object, e As System.EventArgs) Handles comboBox2.SelectedIndexChanged
            Me.chartControl1.ChartArea.InteractiveCursors(0).YInterval = Convert.ToDouble(Me.comboBox2.SelectedItem)
        End Sub
    End Class
End Namespace